# Human Development Index and Components

![](https://www.linkpicture.com/q/map-of-the-world-1005413_1280.jpg)

In this notebook we will analyze the Human Development Index (HDI), life expectancy at birth, expected years of schooling, Gross National Income (GNI), the GNI per capita range minus the HDI range, according to data from the World Human Development Report 2021.

In [1]:
import pandas as pd



In [2]:
with open('Human Development Index and Components.csv') as f:
    print(f)

<_io.TextIOWrapper name='Human Development Index and Components.csv' mode='r' encoding='UTF-8'>


In [3]:

dataset = pd.read_csv('Human Development Index and Components.csv', encoding= 'unicode_escape')


In [4]:
human_development_df = dataset
human_development_df

HDI rank                              Country HUMAN DEVELOPMENT  \
0           1                          Switzerland        VERY HIGH    
1           2                               Norway        VERY HIGH    
2           3                              Iceland        VERY HIGH    
3           4               Hong Kong, China (SAR)        VERY HIGH    
4           5                            Australia        VERY HIGH    
..        ...                                  ...               ...   
190       191                          South Sudan               LOW   
191       192  Korea (Democratic People's Rep. of)            OTHER    
192       193                               Monaco            OTHER    
193       194                                Nauru            OTHER    
194       195                              Somalia            OTHER    

    Human Development Index (HDI)   Life expectancy at birth  \
0                            0.962                      84.0   
1                            0.961                      83.2   
2                            0.959                      82.7   
3                            0.952                      85.5   
4                            0.951                      84.5   
..                             ...                       ...   
190                          0.385                      55.0   
191                             ..                      73.3   
192                             ..                      85.9   
193                             ..                      63.6   
194                             ..                      55.3   

    Expected years of schooling Mean years of schooling  \
0                          16.5                    13.9   
1                          18.2                    13.0   
2                          19.2                    13.8   
3                          17.3                    12.2   
4                          21.1                    12.7   
..                          ...                     ...   
190                         5.5                     5.7   
191                        10.8                      ..   
192                          ..                      ..   
193                        11.7                      ..   
194                          ..                      ..   

    Gross national income (GNI) per capita GNI per capita rank minus HDI rank  \
0                                   66,933                                  5   
1                                   64,660                                  6   
2                                   55,782                                 11   
3                                   62,607                                  6   
4                                   49,238                                 18   
..                                     ...                                ...   
190                                    768                                 -1   
191                                     ..                                 ..   
192                                     ..                                 ..   
193                                 17,730                                 ..   
194                                  1,018                                 ..   

    HDI rank.1  Unnamed: 10  Unnamed: 11  
0            3          NaN          NaN  
1            1          NaN          NaN  
2            2          NaN          NaN  
3            4          NaN          NaN  
4            5          NaN          NaN  
..         ...          ...          ...  
190        191          NaN          NaN  
191         ..          NaN          NaN  
192         ..          NaN          NaN  
193         ..          NaN          NaN  
194         ..          NaN          NaN  

[195 rows x 12 columns]

## Data preparation & cleaning

In [6]:
type(human_development_df)

pandas.core.frame.DataFrame

In [7]:
human_development_df.shape

(195, 12)

Here can view the number of rows (195) and columns (12) in the data frame

Let's view basic information about the data frame.

In [8]:
human_development_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 12 columns):
 #   Column                                  Non-Null Count  Dtype  
---  ------                                  --------------  -----  
 0   HDI rank                                195 non-null    int64  
 1   Country                                 195 non-null    object 
 2   HUMAN DEVELOPMENT                       195 non-null    object 
 3   Human Development Index (HDI)           195 non-null    object 
 4   Life expectancy at birth                195 non-null    float64
 5   Expected years of schooling             195 non-null    object 
 6   Mean years of schooling                 195 non-null    object 
 7   Gross national income (GNI) per capita  195 non-null    object 
 8   GNI per capita rank minus HDI rank      195 non-null    object 
 9   HDI rank.1                              195 non-null    object 
 10  Unnamed: 10                             0 non-null      float6

In [9]:
human_development_df.describe()

HDI rank  Life expectancy at birth  Unnamed: 10  Unnamed: 11
count  195.000000                195.000000          0.0          0.0
mean    97.815385                 71.277949          NaN          NaN
std     56.467551                  7.746484          NaN          NaN
min      1.000000                 52.500000          NaN          NaN
25%     49.500000                 65.700000          NaN          NaN
50%     97.000000                 71.700000          NaN          NaN
75%    146.000000                 76.700000          NaN          NaN
max    195.000000                 85.900000          NaN          NaN

In [10]:
human_development_df.columns

Index(['HDI rank', 'Country', 'HUMAN DEVELOPMENT',
       'Human Development Index (HDI) ', 'Life expectancy at birth',
       'Expected years of schooling', 'Mean years of schooling',
       'Gross national income (GNI) per capita',
       'GNI per capita rank minus HDI rank', 'HDI rank.1', 'Unnamed: 10',
       'Unnamed: 11'],
      dtype='object')

Handle missing, incorrect and invalid data

In [11]:
human_development_df.isnull().sum()

HDI rank                                    0
Country                                     0
HUMAN DEVELOPMENT                           0
Human Development Index (HDI)               0
Life expectancy at birth                    0
Expected years of schooling                 0
Mean years of schooling                     0
Gross national income (GNI) per capita      0
GNI per capita rank minus HDI rank          0
HDI rank.1                                  0
Unnamed: 10                               195
Unnamed: 11                               195
dtype: int64

In [12]:
human_development_df.isna()

HDI rank  Country  HUMAN DEVELOPMENT  Human Development Index (HDI)   \
0       False    False              False                           False   
1       False    False              False                           False   
2       False    False              False                           False   
3       False    False              False                           False   
4       False    False              False                           False   
..        ...      ...                ...                             ...   
190     False    False              False                           False   
191     False    False              False                           False   
192     False    False              False                           False   
193     False    False              False                           False   
194     False    False              False                           False   

     Life expectancy at birth  Expected years of schooling  \
0                       False                        False   
1                       False                        False   
2                       False                        False   
3                       False                        False   
4                       False                        False   
..                        ...                          ...   
190                     False                        False   
191                     False                        False   
192                     False                        False   
193                     False                        False   
194                     False                        False   

     Mean years of schooling  Gross national income (GNI) per capita  \
0                      False                                   False   
1                      False                                   False   
2                      False                                   False   
3                      False                                   False   
4                      False                                   False   
..                       ...                                     ...   
190                    False                                   False   
191                    False                                   False   
192                    False                                   False   
193                    False                                   False   
194                    False                                   False   

     GNI per capita rank minus HDI rank  HDI rank.1  Unnamed: 10  Unnamed: 11  
0                                 False       False         True         True  
1                                 False       False         True         True  
2                                 False       False         True         True  
3                                 False       False         True         True  
4                                 False       False         True         True  
..                                  ...         ...          ...          ...  
190                               False       False         True         True  
191                               False       False         True         True  
192                               False       False         True         True  
193                               False       False         True         True  
194                               False       False         True         True  

[195 rows x 12 columns]

As we can see in this data frame that the data in the columns unnamed 10, and unnamed11 are missing values.

So we are going to eliminate the columns that do not provide any data, such as the column  unname 10, and  unname 11 and creating a new dataframe.

In [13]:
human_develop_df = human_development_df.drop(['Unnamed: 10', 'Unnamed: 11'], axis=1)
human_develop_df

HDI rank                              Country HUMAN DEVELOPMENT  \
0           1                          Switzerland        VERY HIGH    
1           2                               Norway        VERY HIGH    
2           3                              Iceland        VERY HIGH    
3           4               Hong Kong, China (SAR)        VERY HIGH    
4           5                            Australia        VERY HIGH    
..        ...                                  ...               ...   
190       191                          South Sudan               LOW   
191       192  Korea (Democratic People's Rep. of)            OTHER    
192       193                               Monaco            OTHER    
193       194                                Nauru            OTHER    
194       195                              Somalia            OTHER    

    Human Development Index (HDI)   Life expectancy at birth  \
0                            0.962                      84.0   
1                            0.961                      83.2   
2                            0.959                      82.7   
3                            0.952                      85.5   
4                            0.951                      84.5   
..                             ...                       ...   
190                          0.385                      55.0   
191                             ..                      73.3   
192                             ..                      85.9   
193                             ..                      63.6   
194                             ..                      55.3   

    Expected years of schooling Mean years of schooling  \
0                          16.5                    13.9   
1                          18.2                    13.0   
2                          19.2                    13.8   
3                          17.3                    12.2   
4                          21.1                    12.7   
..                          ...                     ...   
190                         5.5                     5.7   
191                        10.8                      ..   
192                          ..                      ..   
193                        11.7                      ..   
194                          ..                      ..   

    Gross national income (GNI) per capita GNI per capita rank minus HDI rank  \
0                                   66,933                                  5   
1                                   64,660                                  6   
2                                   55,782                                 11   
3                                   62,607                                  6   
4                                   49,238                                 18   
..                                     ...                                ...   
190                                    768                                 -1   
191                                     ..                                 ..   
192                                     ..                                 ..   
193                                 17,730                                 ..   
194                                  1,018                                 ..   

    HDI rank.1  
0            3  
1            1  
2            2  
3            4  
4            5  
..         ...  
190        191  
191         ..  
192         ..  
193         ..  
194         ..  

[195 rows x 10 columns]

Seeing that our dataframe is clean of incorrect data, we can start to manipulate it to obtain more specific information.

fist let make a copy of data frame to be able to manipulate it.

In [15]:
human_development = human_development_df.copy()

In [16]:
human_development

HDI rank                              Country HUMAN DEVELOPMENT  \
0           1                          Switzerland        VERY HIGH    
1           2                               Norway        VERY HIGH    
2           3                              Iceland        VERY HIGH    
3           4               Hong Kong, China (SAR)        VERY HIGH    
4           5                            Australia        VERY HIGH    
..        ...                                  ...               ...   
190       191                          South Sudan               LOW   
191       192  Korea (Democratic People's Rep. of)            OTHER    
192       193                               Monaco            OTHER    
193       194                                Nauru            OTHER    
194       195                              Somalia            OTHER    

    Human Development Index (HDI)   Life expectancy at birth  \
0                            0.962                      84.0   
1                            0.961                      83.2   
2                            0.959                      82.7   
3                            0.952                      85.5   
4                            0.951                      84.5   
..                             ...                       ...   
190                          0.385                      55.0   
191                             ..                      73.3   
192                             ..                      85.9   
193                             ..                      63.6   
194                             ..                      55.3   

    Expected years of schooling Mean years of schooling  \
0                          16.5                    13.9   
1                          18.2                    13.0   
2                          19.2                    13.8   
3                          17.3                    12.2   
4                          21.1                    12.7   
..                          ...                     ...   
190                         5.5                     5.7   
191                        10.8                      ..   
192                          ..                      ..   
193                        11.7                      ..   
194                          ..                      ..   

    Gross national income (GNI) per capita GNI per capita rank minus HDI rank  \
0                                   66,933                                  5   
1                                   64,660                                  6   
2                                   55,782                                 11   
3                                   62,607                                  6   
4                                   49,238                                 18   
..                                     ...                                ...   
190                                    768                                 -1   
191                                     ..                                 ..   
192                                     ..                                 ..   
193                                 17,730                                 ..   
194                                  1,018                                 ..   

    HDI rank.1  Unnamed: 10  Unnamed: 11  
0            3          NaN          NaN  
1            1          NaN          NaN  
2            2          NaN          NaN  
3            4          NaN          NaN  
4            5          NaN          NaN  
..         ...          ...          ...  
190        191          NaN          NaN  
191         ..          NaN          NaN  
192         ..          NaN          NaN  
193         ..          NaN          NaN  
194         ..          NaN          NaN  

[195 rows x 12 columns]

Let's working 

In [17]:
human_development.head(10)

HDI rank                 Country HUMAN DEVELOPMENT  \
0         1             Switzerland        VERY HIGH    
1         2                  Norway        VERY HIGH    
2         3                 Iceland        VERY HIGH    
3         4  Hong Kong, China (SAR)        VERY HIGH    
4         5               Australia        VERY HIGH    
5         6                 Denmark        VERY HIGH    
6         7                  Sweden        VERY HIGH    
7         8                 Ireland        VERY HIGH    
8         9                 Germany        VERY HIGH    
9        10             Netherlands        VERY HIGH    

  Human Development Index (HDI)   Life expectancy at birth  \
0                          0.962                      84.0   
1                          0.961                      83.2   
2                          0.959                      82.7   
3                          0.952                      85.5   
4                          0.951                      84.5   
5                          0.948                      81.4   
6                          0.947                      83.0   
7                          0.945                      82.0   
8                          0.942                      80.6   
9                          0.941                      81.7   

  Expected years of schooling Mean years of schooling  \
0                        16.5                    13.9   
1                        18.2                    13.0   
2                        19.2                    13.8   
3                        17.3                    12.2   
4                        21.1                    12.7   
5                        18.7                    13.0   
6                        19.4                    12.6   
7                        18.9                    11.6   
8                        17.0                    14.1   
9                        18.7                    12.6   

  Gross national income (GNI) per capita GNI per capita rank minus HDI rank  \
0                                 66,933                                  5   
1                                 64,660                                  6   
2                                 55,782                                 11   
3                                 62,607                                  6   
4                                 49,238                                 18   
5                                 60,365                                  6   
6                                 54,489                                  9   
7                                 76,169                                 -3   
8                                 54,534                                  6   
9                                 55,979                                  3   

  HDI rank.1  Unnamed: 10  Unnamed: 11  
0          3          NaN          NaN  
1          1          NaN          NaN  
2          2          NaN          NaN  
3          4          NaN          NaN  
4          5          NaN          NaN  
5          5          NaN          NaN  
6          9          NaN          NaN  
7          8          NaN          NaN  
8          7          NaN          NaN  
9         10          NaN          NaN

####  How many countries does the dataframe contain?

In [19]:
human_development.shape

(195, 12)

# Exploratory Analysis and Visualization

Now let's create visualizations of the data frames to get a better idea of the data so we can manipulate it to find information.

In [ ]:
import seaborn as sns
import matplotlib
import matplotlib.pyplot as plt
%matplotlib inline

sns.set_style('darkgrid')
matplotlib.rcParams['font.size'] = 14
matplotlib.rcParams['figure.figsize'] = (9, 5)
matplotlib.rcParams['figure.facecolor'] = '#00000000'